### PR TITLE
Upgrade Elasticsearch from 7.17 to 8

### PR DESCRIPTION
**Pull request by makelicious**
In order to fix the issue https://github.com/opencrvs/opencrvs-core/issues/7149 we needed to upgrade to next major version of elasticsearch.

Changes:
1. Add `{meta: true}` option to keep the payload compatible with 7.x.x format
2. Fix type issues revealed by the upgrade
3. Add search.ts to share type related items between services (migration & search)

Needs config changes from: https://github.com/opencrvs/opencrvs-countryconfig/pull/216

### CODE REVIEW

File: packages/search/src/features/registration/deduplicate/service.ts
Comment (makelicious) [inline packages/search/src/features/registration/deduplicate/service.ts:1]: I would appreciate second pair of eyes especially here to ensure this stayed as it should

File: packages/migration/src/migrations/hearth/20240527130750-reindex-elasticsearch.ts
Comment (Zangetsu101) [inline packages/migration/src/migrations/hearth/20240527130750-reindex-elasticsearch.ts:1]: We should be able to just remove this migration as we are running the elastic reindex on every deploy from here: https://github.com/opencrvs/opencrvs-core/blob/develop/packages/migration/run-migrations.sh#L42

File: packages/migration/src/utils/elasticsearch-helper.ts [hunk 1]
```diff
@@ -23,46 +25,51 @@ export const updateComposition = async (
...
-    response = await client.update({
-      index: ELASTICSEARCH_INDEX_NAME,
-      type: 'compositions',
-      id,
-      body: {
-        doc: body
+    return await client.update(
+      {
+        index: ELASTICSEARCH_INDEX_NAME,
+        // type: 'compositions', @todo: check whether this should work
```
[the remark this answers is not in the document]
Comment (Zangetsu101) [reply packages/migration/src/utils/elasticsearch-helper.ts:32]: The `types` property was removed from the `elasticsearch/dbhelper.ts` file here: https://github.com/opencrvs/opencrvs-core/commit/fc2594d#diff-f1ee21482bf25e832b890c7a5d40825103ec072bf33ec84275d4022b12adfd75L26
So it should safe to do so here as well

File: packages/search/src/features/registration/deduplicate/service.ts [hunk 2]
```diff
@@ -140,14 +147,71 @@ export const searchForBirthDuplicates = async (
...
-      ].filter(Boolean)
+      ].filter(isNonEmptyCondition)
+    }
+  }
+
+  const childsNameMatch = {
+    bool: {
+      must: {
+        bool: {
+          must: [
```
Comment (Zangetsu101) [inline packages/search/src/features/registration/deduplicate/service.ts:158]:
```suggestion
    bool: {
      must: [
```
We probably don't the extra nesting right? It wasn't there previously too

File: packages/search/src/features/registration/deduplicate/service.ts
Comment (Zangetsu101) [reply packages/search/src/features/registration/deduplicate/service.ts:1]: Left a single comment otherwise looked fine to me!

**Review by Zangetsu101 — COMMENTED** [review]
Great job @makelicious! Really appreciate the small refactors here & there

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**